test/compiler/interpreter_exec.jl

```diff
@@ -110,3 +110,29 @@ let m = Meta.@lower 1 + 1
     @test :b === @eval $m
     @test isempty(current_exceptions())
 end
+
+# Test that things don't break if one branch of the frontend PhiNode becomes unreachable
+let m = Meta.@lower 1 + 1
+    @assert Meta.isexpr(m, :thunk)
+    src = m.args[1]::CodeInfo
+    src.code = Any[
+        # block 1
+        GlobalRef(@__MODULE__, :global_error_switch),
+        GotoIfNot(SSAValue(1), 4),
+        # block 2
+        Expr(:call, error, "This error is expected"),
+        # block 3
+        PhiNode(Int32[2, 3], Any[1, 2]),
+        ReturnNode(SSAValue(4))
+    ]
+    nstmts = length(src.code)
+    src.ssavaluetypes = nstmts
+    src.ssaflags = fill(UInt8(0x00), nstmts)
+    src.debuginfo = Core.DebugInfo(:none)
+    Core.Compiler.verify_ir(Core.Compiler.inflate_ir(src))
+    global global_error_switch = true
+    @test_throws ErrorException @eval $m
+    global global_error_switch = false
+    @test 1 === @eval $m
+    @test isempty(current_exceptions())
+end
```

Comment:
The top-level code chunk will be inferred but not optimized. I think we need to do something like below instead in order to really exercise this change:
```julia
function gen_unreachable_phinode_edge_generator(world::UInt, source, _)
    ci = make_codeinfo(Any[
        # block 1
        GlobalRef(@__MODULE__, :global_error_switch),
        GotoIfNot(SSAValue(1), 4),
        # block 2
        Expr(:call, error, "This error is expected"),
        # block 3
        PhiNode(Int32[2, 3], Any[1, 2]),
        ReturnNode(SSAValue(4))
    ]; slottypes=Any[Any])
    ci.slotnames = Symbol[:var"#self#"]
    ci
end

@eval function gen_unreachable_phinode_edge()
    $(Expr(:meta, :generated, gen_unreachable_phinode_edge_generator))
    $(Expr(:meta, :generated_only))
    #= no body =#
end

global global_error_switch = true
@test_throws ErrorException gen_unreachable_phinode_edge()
global global_error_switch = false
@test gen_unreachable_phinode_edge() == 1
```

Reply:
That should work, but I was pretty sure I saw the error in question with this test.

Reply:
I just edited my local interpreter_exec.jl to include the new test case, and got the following run successfully:
```julia
shell> cd .
~/julia4/test/compiler

julia> module CompilerExecTest
       include("interpreter_exec.jl")
       end
Main.CompilerExecTest

julia> module InterpreterExecTest
       # test code execution with the interpreter mode (compile=min)
       Base.Experimental.@compiler_options compile=min
       include("interpreter_exec.jl")
       end
Main.InterpreterExecTest
```

Reply:
Yeah, leave it to me.